xbmc/interfaces/legacy/ListItem.cpp

```diff
@@ -396,6 +398,21 @@ namespace XBMCAddon
           else if (key == "dateadded")
             item->GetVideoInfoTag()->m_dateAdded.SetFromDBDateTime(value.c_str());
         }
+
+        // For backward compatibility.
+        // FIXME: Remove this behaviour. It should be possible to set only tvshowtitle without
+        // having mediatype implicitly changed.
+        if (item->GetVideoInfoTag()->m_type == MediaTypeNone)
+        {
+          if (!item->GetVideoInfoTag()->m_strShowTitle.empty() && item->GetVideoInfoTag()->m_iSeason == -1)
+          {
+            item->GetVideoInfoTag()->m_type = MediaTypeTvShow;
+          }
+          else if (!item->GetVideoInfoTag()->m_artist.empty())
+          {
+            item->GetVideoInfoTag()->m_type = MediaTypeMusicVideo;
+          }
```

Comment:
@Montellese Ah, forgot about episode, added! Not sure what to do about movies though. None of the checks changed here are for movies, so what would the magic value for falling back to MediaTypeMovie be?